Bluetooth: Controller: Fix SID for legacy advertising reported

cvinayak opened the pull request:
SID value for legacy advertising reported shall be 255.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix SID for legacy advertising reported

SID value for legacy advertising reported shall be 255.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (approved):
Would appreciate a #define for this value as well, but it's okay to merge without.

cvinayak committed:
Bluetooth: Add define for invalid Extended Advertising SID value

Add a define for invalid Extended Advertising SID value of
255 required when generating legacy advertising report.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
